smart-connector-rest-server/src/main/java/eu/knowledge/engine/rest/api/impl/ProactiveApiServiceImpl.java

```diff
@@ -145,9 +149,16 @@ public void scAskPost(
 								.failedMessage(aei.getFailedMessage()))
 						.collect(Collectors.toList());
 
-				AskResult ar = new AskResult().bindingSet(this.bindingSetToList(askResult.getBindings()))
-						.exchangeInfo(infos);
+				LOG.debug("Bindings in result is {}", askResult.getBindings());
+				AskResult ar = new AskResult().bindingSet(this.bindingSetToList(askResult.getBindings())).exchangeInfo(infos);
 
+				LOG.debug("KnowledgeGapsEnabled is {}", ki.getKnowledgeGapsEnabled());
+				// add knowledge gaps if knowledgeGapsEnabled is true
+				if (ki.getKnowledgeGapsEnabled()) {
```

Comment:
So, I had to rethink what is going on here. We need to distinguish between the following situations:

1. The KI is not knowledgeGapsEnabled => no knowledge gaps will be returned
2. The KI is knowledgeGapsEnabled AND there are bindings found => knowledge gaps will be empty
3. The KI is knowledgeGapsEnabled AND there are no bindings found => knowledge gaps will contain the gaps found (by the way, might be empty as well).

The above code does not cover situation 2 well as there was a null result for the knowledge gaps instead of an empty one.

Reply:
Hi @JackJackie, as discussed yesterday I agree that the above situations are indeed how we would like the REST API te respond 👍 

Were you already able to implement this? Currently, the `TestAskWithGapsNotEnabled#testAskWithoutGaps` unit test fails because it returns knowledge gaps even when they are not enabled:

```
Error: [ERROR] eu.knowledge.engine.rest.api.TestAskWithGapsNotEnabled.testAskWithoutGaps -- Time elapsed: 1.468 s <<< FAILURE!
org.opentest4j.AssertionFailedError: expected: <false> but was: <true>
	at org.junit.jupiter.api.AssertionFailureBuilder.build(AssertionFailureBuilder.java:151)
	at org.junit.jupiter.api.AssertionFailureBuilder.buildAndThrow(AssertionFailureBuilder.java:132)
	at org.junit.jupiter.api.AssertFalse.failNotFalse(AssertFalse.java:63)
	at org.junit.jupiter.api.AssertFalse.assertFalse(AssertFalse.java:36)
	at org.junit.jupiter.api.AssertFalse.assertFalse(AssertFalse.java:31)
	at org.junit.jupiter.api.Assertions.assertFalse(Assertions.java:231)
	at eu.knowledge.engine.rest.api.TestAskWithGapsNotEnabled.testAskWithoutGaps(TestAskWithGapsNotEnabled.java:64)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
```

Reply:
I've fixed one of the REST API unit tests by making sure the ordering of triples within a knowledge gap are not important for the assertion. It failed for me, because the ordering was different.

Reply:
Yes, I noticed that as well....now I'm dealing with the fact that the AskResult in the RestAPI needs to give back 2 different objects as part of the '200' response....one with knowledgeGaps and one without....to be continued!!

Reply:
Did take care of it...check it out!

Reply:
Good to see you got it sorted out 👍 , although having an additional `AsjResultWithGaps` seems a bit overkill. Maybe we could also try this solution? 

https://stackoverflow.com/a/72734037

Not sure if it works and it might break other things, but configuring it seems like a 'cleaner' solution. What do you think?